code/implementation/system-stack/src/main/java/io/cattle/platform/systemstack/listener/external/api/catalog/Template.java

```diff
@@ -0,0 +1,376 @@
+package io.cattle.platform.systemstack.listener.external.api.catalog;
+
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class Template {
+    @JsonProperty("catalogId")
+    private String catalogId;
+
+    @JsonProperty("category")
+    private String category;
+
+    @JsonProperty("defaultVersion")
+    private String defaultVersion;
+
+    @JsonProperty("description")
+    private String description;
+
+    @JsonProperty("id")
+    private String id;
+
+    @JsonProperty("license")
+    private String license;
+
+    @JsonProperty("maintainer")
+    private String maintainer;
+
+    @JsonProperty("minimumRancherVersion")
+    private String minimumRancherVersion;
+
+    @JsonProperty("name")
+    private String name;
+
+    @JsonProperty("path")
+    private String path;
+
+    @JsonProperty("type")
+    private String type;
+
+    @JsonProperty("versionLinks")
+    private Map<String, String> versionLinks;
+
+    @JsonProperty("dockerCompose")
+    private String dockerCompose;
+
+    @JsonProperty("rancherCompose")
+    private String rancherCompose;
+
+    @JsonProperty("uuid")
+    private String uuid;
+
+    @JsonProperty("version")
+    private String version;
+
+    /**
+    * 
+    * @return
+    * The catalogId
+    */
+    @JsonProperty("catalogId")
+    public String getCatalogId() {
+    return catalogId;
+    }
+
+    /**
```

Comment:
Remove all the javadocs for getters/setters.  They don't provide any value unless they have some meaningful info in them.